packages/client/test/integration/lightsync.spec.ts

```diff
@@ -0,0 +1,75 @@
+import tape from 'tape'
+import { wait, setup, destroy } from './util'
+
+tape('[Integration:LightSync]', async (t) => {
+  t.test('should sync headers', async (t) => {
+    const [remoteServer, remoteService] = await setup({
+      location: '127.0.0.2',
+      height: 200,
+      syncmode: 'full',
+    })
+    const [localServer, localService] = await setup({
+      location: '127.0.0.1',
+      height: 0,
+      syncmode: 'light',
+    })
+    localService.on('synchronized', async () => {
+      t.equals(localService.chain.headers.height.toNumber(), 200, 'synced')
+      await destroy(localServer, localService)
+      await destroy(remoteServer, remoteService)
+      t.end()
+    })
+    await localServer.discover('remotePeer', '127.0.0.2')
+  })
+
+  t.test('should not sync with stale peers', async (t) => {
```

Comment:
Just a side note: This test passes in CI but is also throwing an unhandled promise rejection warning.

Reply:
I think this originally also threw in the client tests. But certainly worth looking into at some point.

Reply:
Added to #975